torchvtk/augmentation/DictTransform.py

```diff
@@ -0,0 +1,150 @@
+                    # transform
+                    batch.append(self.transform(tmptensor))
+                # stack tensor back
+                tmp = torch.stack(batch, dim=0)
+
+            if self.dtype is torch.float16:
+                tmp = tmp.to(torch.float16)
+            elif self.dtype is not torch.float16 or torch.float32:
+                tmp = tmp.to(torch.float32)
+            tmp = tmp.to(self.device)
+            # back on device?
+            data[key] = tmp
+        return data
+
+
+class NoiseDictTransform(DictTransform):
+
+    def __init__(self, device, noise_variance=(0.001, 0.05), apply_on=["vol"], dtype=torch.float32, batch_transform=False):
+
+        self.device = device
+        self.noise_variance = noise_variance
+        DictTransform.__init__(self, self.device, apply_on=apply_on, dtype=dtype, batch_transform=batch_transform)
+
+    def transform(self, data):
+        #todo allow for batch
+        variance = random.uniform(self.noise_variance[0], self.noise_variance[1])
+        noise = np.random.normal(0.0, variance, size=data.shape)
+        noise = torch.from_numpy(noise)
+        noise = noise.to(self.device)
+        sample = torch.add(data, noise)
+        return sample
+
+
+class BlurDictTransform(DictTransform):
+
+    def __init__(self, apply_on=["vol"], dtype=torch.float32, device= "cpu", channels=1, kernel_size=(3, 3, 3), sigma=1, batch_transform=False):
+        """
+
+        :param device:
+        :param channels:
+        :param kernel_size:
+        :param sigma:
+        """
+        self.channels = channels
+        self.sigma = [sigma, sigma, sigma]
+        self.kernel_size = kernel_size
+        self.device = device
+        DictTransform.__init__(self, self.device, apply_on=apply_on, dtype=dtype, batch_transform=batch_transform)
+        # code from: https://discuss.pytorch.org/t/is-there-anyway-to-do-gaussian-filtering-for-an-image-2d-3d-in-pytorch/12351/10
+        # initialize conv layer.
+        kernel = 1
+        # todo add dynamic padding for higher kerner_size
+        #
+        self.meshgrids = torch.meshgrid(
+            [
+                torch.arange(size, dtype=torch.float32)
+                for size in self.kernel_size
+            ]
+        )
+        for size, std, mgrid in zip(self.kernel_size, self.sigma, self.meshgrids):
+            mean = (size - 1) / 2
+            kernel *= 1 / (std * math.sqrt(2 * math.pi)) * torch.exp(-((mgrid - mean) / std) ** 2 / 2)
+
+        # Make sure sum of values in gaussian kernel equals 1.
+        kernel = kernel / torch.sum(kernel)
+
+        # Reshape to depthwise convolutional weight
+        kernel = kernel.view(1, 1, *kernel.size())
+        kernel = kernel.repeat(self.channels, *[1] * (kernel.dim() - 1))
+
+        # todo check if that method of assigning works
+        # self.register_buffer('weight', kernel)
+        kernel = kernel.to(self.device)
+        self.weight = kernel
+        self.groups = self.channels
```

Comment:
what does the channels thing do? I suppose you want to blur each volume channel separately? I would suggest better getting the channels from the input tensor then. After `make_5d`, `tensor.size(1)` should always be correct